tools/clang/lib/SPIRV/NonUniformVisitor.cpp

```diff
@@ -0,0 +1,61 @@
+//===--- NonUniformVisitor.cpp - NonUniform Visitor --------------*- C++ -*-==//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
```

Comment:
By only reading [the spec](https://github.com/KhronosGroup/SPIRV-Registry/blob/master/extensions/EXT/SPV_EXT_descriptor_indexing.asciidoc), I was not able to quickly understand what instructions we have to propagate `NonUniformEXT`.

This class only propagates it for `OpLoad`, `OpAccessChain`, ..., and atomic instructions.

Could you please give a description of the reasoning behind this decision in this class?
In addition, the reference about `NonUniformEXT` (e.g., like the link to the spec you read) might be useful.